packages/renderer/src/lib/ui/KubernetesCurrentContextConnectionBadge.svelte

```diff
@@ -15,9 +21,15 @@ function getClassColor(state?: ContextGeneralState): string {
 }
 
 $: text = getText($kubernetesCurrentContextState);
+$: currentContextName = $kubernetesContexts?.find(c => c.currentContext)?.name;
 </script>
 
 {#if $kubernetesCurrentContextState}
-  <Label role="status" name={text} tip={$kubernetesCurrentContextState.error}
-    ><div class="w-2 h-2 {getClassColor($kubernetesCurrentContextState)} rounded-full mx-1"></div></Label>
+  <div class="flex items-center flex-row">
+    {#if !!currentContextName && $kubernetesContextsCheckingStateDelayed?.get(currentContextName)}
+      <div class="mr-1"><Spinner size="12px"></Spinner></div>
+    {/if}
+    <Label role="status" name={text} tip={$kubernetesCurrentContextState.error}
+      ><div class="w-2 h-2 {getClassColor($kubernetesCurrentContextState)} rounded-full mx-1"></div></Label>  
+  </div>
```

Comment:
We should not display both the spinner and the status, should be either loading, either status, what do you think ?

Reply:
I was thinking the same thing, also why not just edit status button to be in-progress similar to our button UI?

Reply:
I suppose there are many ways to do it. For now, I tried to make something very similar to what is in the Settings > Kubernetes page.

Reply:
@ekidneyrh could you please have a look at this design?

Reply:
Maybe this should be on the empty screen where the focus is instead? We've also talked about a Try Again / Refresh Now option in the future, which would fit better there.

That said, it is better to merge this sprint (it's useful now; might get feedback) vs ideal UX, so if there is no immediate agreement on better & easily implementable I would approve as-is and we can continue discussion.

Reply:
Right, I started to work on this refresh button in https://github.com/containers/podman-desktop/pull/9491

I wanted to have this indicator that the check is occurring, before to add a refresh button, which could help the user understand better what happens